AutoRest/Generators/CSharp/Azure.CSharp.Tests/PageJsonTest.cs

```diff
@@ -73,39 +72,11 @@ public void TestNextLinkDeSerialization()
                 ContractResolver = new ReadOnlyJsonContractResolver()
             };
             deserializeSettings.Converters.Add(new ResourceJsonConverter());
-            deserializeSettings.Converters.Add(new PolymorphicDeserializeJsonConverter<Product>("dType"));
-            var deserializedProduct = JsonConvert.DeserializeObject<Page<Product>>(responseBody, deserializeSettings);
+            deserializeSettings.Converters.Add(new PolymorphicDeserializeJsonConverter<TestProduct>("dType"));
+            var deserializedProduct = JsonConvert.DeserializeObject<Fixtures.Azure.AcceptanceTestsPaging.Models.Page<TestProduct>>(responseBody, deserializeSettings);
```

Comment:
Could you change this to be IPage rather than the concrete implementation of the generated class. I'm thinking that IPage should live in the client runtime and also implement IEnumerable. IPage would consist of the enumerator and the NextLink, but not be the concrete implementation. Leaving it to the generator to implement the interface.

Thoughts?